jsf-ri/src/main/java/com/sun/faces/spi/ServiceFactory.java

```diff
@@ -86,11 +86,8 @@ static Object getProviderFromEntry(String entry, Class<?>[] argumentTypes, Objec
                 throw new FacesException("Unable to find constructor accepting arguments: " + Arrays.toString(arguments));
             }
             return c.newInstance(arguments);
-        } catch (ClassNotFoundException | NoSuchMethodException | SecurityException | FacesException | InstantiationException | IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
-            if (LOGGER.isLoggable(Level.FINE)) {
-                LOGGER.log(Level.FINE, e.toString(), e);
-            }
-            return null;
+        } catch (ClassNotFoundException  | SecurityException | InstantiationException | IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
```

Comment:
I think we should leave the ```NoSuchMethodException``` here and just wrap it in a ```FacesException```.

Reply:
Because I want handle `NoSuchMethodException` differently for one and two arguments in AnnotationProviderFactory. Anyway I think I can still get the exception cause from FacesException there and decide if it should be logged.